File: cpp/1814. 统计一个数组中好对子的数目.cpp
class Solution {
public:
    int countNicePairs(vector<int>& nums) {
        unordered_map<int,int> h;
        int MOD=1e9+7;
        int res=0;
        for(int i:nums){
            int t=i,j=0;
            while(t){
                j=j*10+t%10;
                t/=10;
            }
            res=(res+h[j-i])%MOD;
            h[j-i]++;
        }
        return res;
    }
};

File: 刷题/golang/1814. 统计一个数组中好对子的数目.go
func countNicePairs(nums []int) int {
    h:=map[int]int{}
    ans:=0
    for _,x:=range nums{
        t,j:=x,0
        for ;t>0;t/=10{
            j=j*10+t%10
        }
        ans=(ans+h[x-j])%(1e9+7)
        h[x-j]++
    }
    return ans
}